api/app/main.py

```diff
@@ -1,7 +1,46 @@
-from fastapi import FastAPI
+from typing import Optional
+from fastapi import FastAPI, Request, Header
+from fastapi.responses import HTMLResponse
+from fastapi.templating import Jinja2Templates
+import requests
+from PIL import Image
+from io import BytesIO
+import numpy as np
+import matplotlib.pyplot as plt
+import json
 
+#uvicorn main:app --reload
 app = FastAPI()
+templates = Jinja2Templates(directory="./HTML")
 
-@app.get("/")
-def root():
-    return {"hello": "world"}
+@app.get("/", response_class=HTMLResponse)
+def root(request: Request):
+    return templates.TemplateResponse("homepage.html", {"request": request}) #just a useless homepage.XD
+
+@app.get("/image/")
+def url_post(request: Request):
+    return templates.TemplateResponse("showimag.html", {"request": request,"test":request.headers}) #the HTML file can send requests with image url in the headers. using XMLHttpRequests
```

Comment:
I prefer to keep the front end and back end code separate, and avoid blending the two as much as possible. No need to remove this for now though.

Maybe in another session we can look at how to send requests to the API from the cli or from another python script...